components/modules/rmt.c

```diff
@@ -112,6 +112,13 @@ static int lrmt_txsetup(lua_State *L) {
       config.flags |= RMT_CHANNEL_FLAGS_INVERT_SIG;
     }
     lua_pop(L, 1);
+
+    lua_getfield(L, 3, "slots");
+    int slots = lua_tointeger(L, -1);
```

Comment:
Won't this raise an error if `slots` is `nil`? Consider using `luaL_optinteger()`

Reply:
lua_tointeger returns 0 if the value cannot be converted to an integer and this is checked in the next line